### ISSUE TITLE
linear-gradient overlaying an image?

### ISSUE THREAD

**Issue by mhigley** [timhagn/gatsby-background-image#39]
@timhagn ...

I'm loving the package. It's made adding in background images SO easy!! And optimized!! 👏  But for many of my Hero images, I'm needing to overlay a gradient for text readability. I'm trying your multiple images technique which looks like it should be working, except the inspector is showing the gradient is base64ing into a 1x1px gif. I'm I trying something impossible?

Here are select relevant snippets of code: 
```
const {
    heroBackground,
    heroGradient = `linear-gradient(hsla(0, 0%, 0%, .5), hsla(0, 0%, 0%, .1))`,
  } = useStaticQuery ...
```
&
```
const imageStack = [
    heroBackground.childImageSharp.fluid,
    heroGradient,
  ].reverse();
```

And here's what the inspector is showing:
```
.Hero__HeroWrapper-hBceBM.dyEfMt::before {
    z-index: -100;
    background-image: url(data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==),url(http://localhost:8000/static/d9ab300c84889f385d02e740a8ee3b94/d5475/hero.webp);
    opacity: 1;
}
```
Any advice? Am I missing something?
Thanks!

Matt

**Comment by timhagn** [comment]
Hi Matt / @mhigley,

I'm wondering why Gatsby doesn't bail out on your first snippet in the first place - or do you query for other things (e.g. an image) than gradient strings in GraphQL before destructuring the result of the `useStaticQuery` hook?

The rest of it is currently working "correctly" in `gbi`, as your `heroGradient` is neither a `fluid` nor a `fixed` Object (as you get em from `gatsby-transformer-sharp` and similar plugins), which are the only types usable in arrays given to `gbi` (am still wondering why Gatsby doesn't bail on the mismatched type ; ) at the moment. 
The 1x1px transparent gif is a placeholder (the smallest cross-browser one I could find ^^) preventing other images in the array not getting the correct `background-position` and suchlike, lest a wild jumping around might happen (as it did happen for me before `v0.7.1`).

Your issue is nearly a duplicate of #38, where `rgba()` would be needed instead of a `linear-gradient`.
As I already mentioned in a comment [over at it](https://github.com/timhagn/gatsby-background-image/issues/38#issuecomment-502433572), I'm gonna try to find the time to implement string handling for cases like this, though a large part of the image handling functions need additions / rewriting for it to work.

The `rgba()` opacity issue might be worked around with setting an inner wrapper with an absolute position like I mentioned in my comment before the one linked above, but I ain't sure if this would work with a background gradient (have tried neither).

Gonna keep you posted on my progress (and help would of course be greatly appreciated % ),

greetz,

Tim.

**Comment by timhagn** [comment]
Hi again, Matt / @mhigley!

As you were already the third one wanting this behavior, I quickly had a look. 
And as it turned out, this was far easier to implement than I imagined! 
It is now possible (at least it is at my place with FF and Chrome % ) with the just published `v0.7.2` like shown in the example in its [README](https://github.com/timhagn/gatsby-background-image/tree/master/packages/gatsby-background-image#how-to-use-with-multiple-images) : ).
Your snippets should now function as wanted, too.

Should it work for you, feel free to have the honor to close this issue : ).

Best,

Tim.

**Comment by timhagn** [comment]
As this issue should be solved, seems stale for some time and ProBot strangely didn't act, I'm closing it now manually. Should there still arise problems, just give a shout.
